macros/get_lookback_date_limits.sql

```diff
@@ -36,19 +42,19 @@
 
   {% set query %}
     {% if limit_type == 'min' %}
-      with base as (select case when '{{ var("snowplow__conversion_window_start_date") }}' = ''
-                  then {{ dbt.dateadd('day', -31, dbt.current_timestamp()) }}
-                  else '{{ var("snowplow__conversion_window_start_date") }}'
-                  end as min_date_time)
-      select cast({{ dbt.dateadd('day', (- var('snowplow__path_lookback_days') + 1), 'min_date_time') }} as date) from base
-
+      {% if var("snowplow__conversion_window_start_date") == '' %}
+        with base as (select {{ dbt.dateadd('day', -var("snowplow__conversion_window_days"), dbt.current_timestamp()) }} as min_date_time)
+        select cast({{ dbt.dateadd('day', (- var('snowplow__path_lookback_days') + 1), 'min_date_time') }} as date) from base
+      {% else %}
+        select cast('{{ var("snowplow__conversion_window_start_date") }}' as date)
+      {% endif %}
```

Comment:
You still need to add the path lookback days do you not? Or was that wrong in the original approach?

Reply:
I think what caused the confusion originally even is that the lookback date limit macro is used for both the conversions and the sessions (pageviews) table. I think the conversions don't need the lookback days and therefore what we have currently is wrong.

Reply:
I reworked the whole macro taking the above discovery in mind (aka adding another parameter to define which model to get the limits for and applying different logic) and added a unit test to help us figure out how the current logic works (it actually helped me debug along the way), let me know if it is not clear, I tried to be explicit with comments with a use case to demonstrate it in the unit test actual file.